internal/server/repository_worker_test.go

```diff
@@ -691,12 +695,36 @@ func TestListWorkers_WithExcludeShutdown(t *testing.T) {
 		server.NewWorker(scope.Global.String(),
 			server.WithName(worker3.GetName()),
 			server.WithAddress(worker3.GetAddress()),
+			server.WithReleaseVersion("Boundary v.0.11"),
 			server.WithOperationalState(server.ShutdownOperationalState.String())),
 		server.WithPublicId(worker3.GetPublicId()))
 	require.NoError(err)
 	result, err = serversRepo.ListWorkers(ctx, []string{scope.Global.String()}, server.WithExcludeShutdownWorkers(true))
 	require.NoError(err)
 	require.Len(result, 0)
+
+	// Upsert without a release version (nil version) or state and expect to get a hit- test backwards compatibility
+	_, err = serversRepo.UpsertWorkerStatus(ctx,
+		server.NewWorker(scope.Global.String(),
+			server.WithName(worker3.GetName()),
+			server.WithAddress(worker3.GetAddress())),
+		server.WithPublicId(worker3.GetPublicId()))
+	require.NoError(err)
+	result, err = serversRepo.ListWorkers(ctx, []string{scope.Global.String()}, server.WithExcludeShutdownWorkers(true))
+	require.NoError(err)
+	require.Len(result, 1)
+
+	// Upsert with an empty string release version and no state
+	_, err = serversRepo.UpsertWorkerStatus(ctx,
+		server.NewWorker(scope.Global.String(),
+			server.WithName(worker3.GetName()),
+			server.WithAddress(worker3.GetAddress()),
+			server.WithReleaseVersion("")),
+		server.WithPublicId(worker3.GetPublicId()))
+	require.NoError(err)
+	result, err = serversRepo.ListWorkers(ctx, []string{scope.Global.String()}, server.WithExcludeShutdownWorkers(true))
```

Comment:
I'm not sure we are testing all of the combinations in the `where` clause. These are the ones I came up with, but please let me know if I'm wrong:

(`¯\_(ツ)_/¯` represents a `null` value, and the big blank space is a string with spaces)

```
test case │ operational_state │ release_version │ return?
══════════╪═══════════════════╪═════════════════╪═══════════
     1    │ active            │ Boundary v.0.11 │  True
     2    │ active            │ ¯\_(ツ)_/¯      │  True
     3    │ active            │                 │  True
     4    │ shutdown          │ Boundary v.0.11 │  False
     5    │ shutdown          │ ¯\_(ツ)_/¯      │  True
     6    │ shutdown          │                 │  True
```

I queried the test database right before each call to `serversRepo.ListWorkers` and got these results:

```
select public_id, operational_state, release_version from server_worker;

At line 663:
  public_id   │ operational_state │ release_version    test case
══════════════╪═══════════════════╪══════════════════
 w_13VWxoTRI1 │ active            │ Boundary v0.11.0   1
 w_lsFI8gEYUC │ active            │ Boundary v0.11.0   1
 w_BOns6xVfAW │ active            │ Boundary v0.11.0   1

At line 678:
  public_id   │ operational_state │ release_version
══════════════╪═══════════════════╪══════════════════
 w_B1LF31hSjA │ active            │ Boundary v0.11.0   1
 w_wJmlVvxp8v │ active            │ Boundary v0.11.0   1
 w_lS4zt6IwAR │ shutdown          │ Boundary v.0.11    4

At line 690:
  public_id   │ operational_state │ release_version
══════════════╪═══════════════════╪══════════════════
 w_i6wxYZZcjY │ active            │ Boundary v0.11.0   1
 w_bKS63cjh6T │ shutdown          │ Boundary v.0.11    4
 w_v5gYvY3nAz │ shutdown          │ Boundary v.0.11    4

At line 702:
  public_id   │ operational_state │ release_version
══════════════╪═══════════════════╪═════════════════
 w_x8iaBdptE9 │ shutdown          │ Boundary v.0.11    4
 w_Re8qJcxg7i │ shutdown          │ Boundary v.0.11    4
 w_vO92o9ikUp │ shutdown          │ Boundary v.0.11    4

At line 713:
  public_id   │ operational_state │ release_version
══════════════╪═══════════════════╪═════════════════
 w_x8iaBdptE9 │ shutdown          │ Boundary v.0.11    4
 w_Re8qJcxg7i │ shutdown          │ Boundary v.0.11    4
 w_vO92o9ikUp │ active            │ ¯\_(ツ)_/¯         2

At line 725:
  public_id   │ operational_state │ release_version
══════════════╪═══════════════════╪═════════════════
 w_KiJjHwNUDl │ shutdown          │ Boundary v.0.11    4
 w_Oh5yP6wVhv │ shutdown          │ Boundary v.0.11    4
 w_9BbPwH6zPk │ active            │ ¯\_(ツ)_/¯         2
```
So we have tests for test cases 1, 2, and 4, but not for 3, 5, and 6.

Reply:
Ahh thanks! Old workers (pre 0.11.0) won't report operational state or release version and will default to Active, so cases 5 and 6 can't happen but that lead me to look at other cases... 

Old and new workers will have an Unknown state with no version on Create and will only be updated after a status update. New workers will report state/ version on status and old workers will default to active after status. A controller-led worker would be in an Unknown state/ without version until a user starts it & and it sends status, so we wouldn't want to send it work.

I've updated the where-clause to only include active workers and updated the test cases to cover these.

Reply:
I think it would be helpful to convert the existing tests to a table-driven test structure. That would allow each test to have a name and it would make it easier to add new ones. What do you think?

Reply:
Sure, I'll do that!